include/exadg/operators/operator_base.cpp

```diff
@@ -2153,6 +2155,173 @@ OperatorBase<dim, Number, n_components>::evaluate_face_integrals() const
          (integrator_flags.face_integrate != dealii::EvaluationFlags::nothing);
 }
 
+template<int dim, typename Number, int n_components>
+void
+OperatorBase<dim, Number, n_components>::compute_factorized_as_matrices() const
+{
+#ifdef DEAL_II_WITH_TRILINOS
+  internal_compute_factorized_as_matrices<dealii::TrilinosWrappers::SparseMatrix>();
+#elif DEAL_II_WITH_PETSC
+  internal_compute_factorized_as_matrices<dealii::PETScWrappers::MPI::SparseMatrix>();
+#else
+  AssertThrow(n_mpi_processes == 1,
+              ExcMessage(
+                "Use Trilinos or Petsc for distributed sparse matrices needed for this function!"));
+  internal_compute_factorized_as_matrices<dealii::SparseMatrix>();
+#endif
+}
+
+template<int dim, typename Number, int n_components>
+template<typename SparseMatrix>
+void
+OperatorBase<dim, Number, n_components>::internal_compute_factorized_as_matrices() const
+{
+  dealii::DoFHandler<dim> const & dof_handler =
+    this->matrix_free->get_dof_handler(this->data.dof_index);
+
+  unsigned int const dofs_per_cell = matrix_free->get_dofs_per_cell(this->data.dof_index);
+
+  unsigned int const n_cells = matrix_free->n_cell_batches() * vectorization_length;
+
+  // assemble a temporary sparse matrix to cut out the blocks
+  SparseMatrix                   tmp_matrix;
+  dealii::DynamicSparsityPattern dsp;
+  internal_init_system_matrix(tmp_matrix, dsp, dof_handler.get_communicator());
+  internal_calculate_system_matrix(tmp_matrix);
+
+  // cut out overlapped block matrices
+  std::vector<dealii::FullMatrix<Number>> overlapped_block_matrix(
```

Comment:
I would use "matrices" here since it is a vector of matrices.

Reply:
And this is also changed in 0946e0ce697c568d48f462918088c27ec907664c